test/controllers/api/v2/subscriptions_controller_test.rb

```diff
@@ -63,6 +64,15 @@ def test_index_protected
       end
     end
 
+    def test_index_with_activation_key_id
+      activation_key = ActivationKey.new
+      Katello::ActivationKey.expects(:find_by).returns(activation_key)
+      Katello::Pool::ActiveRecord_Relation.any_instance.expects(:for_activation_key).returns(Pool.readable)
+      get :index, activation_key_id: 1
+      assert_response :success
+      assert_template 'api/v2/subscriptions/index'
```

Comment:
I would rather we tested that the controller is returning the proper results instead of trusting the model. Is there a way to do this?